packages/webdriverio/src/commands/element/submit.js

```diff
@@ -0,0 +1,43 @@
+/**
+ *
+ * Submits a form found by given selector. The submit command may also be applied
+ * to any element that is a descendant of a `<form>` element.
+ *
+ * <example>
+    :index.html
+    <form action="/form.php" method="post" id="loginForm">
+        <label for="username">User:</label>
+        <input type="text" name="username" id="username">
+        <label for="password">Password:</label>
+        <input type="password" name="password" id="password">
+        <input type="submit" value="Login">
+    </form>
+    :submitForm.js
+    it('should submit login form', function () {
+        browser.setValue('#username', 'foobar');
+        browser.setValue('#password', 'test123');
+        browser.submitForm('#loginForm');
+    });
+ * </example>
+ *
+ * @alias browser.submitForm
+ * @param {String} selector form element
+ * @uses protocol/element, protocol/submit
+ * @type action
+ *
+ */
+
+export default async function submit () {
+    /**
+     * if not WebDriver protocol use jsonwp which has a submit command
+     */
+    if (!this.isW3C) {
+        return this.elementSubmit(this.elementId)
+    }
+
+    /**
+     * otherwise find submit button within form (this scope) and click it
+     */
+    const submitBtn = await this.$('*[type="submit"]')
+    return submitBtn.click()
```

Comment:
I think the docs in this file should probably include that this is how the method can work... I've seen way too much strange HTML out there and would imagine someone trying to use submitForm w/o understanding it would run in to issues with their form submission. 

Maybe something along the lines of (assuming I'm correct that `elementSubmit` is deprecated):
This function works by either trying the deprecated `elementSubmit` command, or by finding the 'submit' element (any element with a `type` attribute of `submit`) and clicking it.

Reply:
I agree with you @klamping as I believe this method shouldn't even exist. It's already possible to click a button of the type submit with the default click method. So preferably to keep everything clean I would actually like to ask if this method is still mandatory :)

Reply:
I agree, the submit command is not really useful at all. I will remove it.